canvas/biplot.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import division
+import matplotlib
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+import numpy as np
+import pandas as pd
+from skbio.stats.composition import clr, closure, centralize,
+from skbio.stats.composition import multiplicative_replacement, perturb_inv
+from biom import load_table
+from functools import partial
+import math
+
+
+def make_biplot(sample_df,
+                feature_df,
+                sample_df_metadata=None,
+                feature_df_metadata=None,
+                sample_color_category=None,
+                feature_color_category=None,
+                sample_color_dict=None,
+                feature_color_dict=None,
+                eigenvalue=None):
+
+    if not isinstance(sample_df, pd.core.frame.DataFrame):
+        raise ValueError('`sample_df` must be a `pd.DataFrame`, '
+                         'not %r.' % type(sample_df).__name__)
+
+    if not isinstance(feature_df, pd.core.frame.DataFrame):
+        raise ValueError('`feature_df` must be a `pd.DataFrame`, '
+                         'not %r.' % type(feature_df).__name__)
+
+    if sample_df_metadata is not None:
+        if not isinstance(sample_df_metadata, pd.core.frame.DataFrame):
+            raise ValueError('`sample_df_metadata` must be a `pd.DataFrame`, '
+                             'not %r.' % type(sample_df_metadata).__name__)
+
+    if feature_df_metadata is not None:
+        if not isinstance(feature_df_metadata, pd.core.frame.DataFrame):
+            raise ValueError('`feature_df_metadata` must be a `pd.DataFrame`, '
+                             'not %r.' % type(feature_df_metadata).__name__)
+
+    # if sample_color_dict is None.
+    # Specify a default palette (default colors).
+    # if (sample_color_dict == None)
+
+    if sample_color_dict is not None:
+        if not isinstance(sample_color_dict, dict):
+            raise ValueError('`sample_color_dict` must be a `dictionary`, '
+                             'not %r.' % type(sample_color_dict).__name__)
+
+    # if feature_color_dict is None.
+    # Specify a default palette (default colors).
+    # if (feature_color_dict == None):
+    if feature_color_dict is not None:
+        if not isinstance(feature_color_dict, dict):
+            raise ValueError('`feature_color_dict` must be a `dictionary`, '
+                             'not %r.' % type(feature_color_dict).__name__)
+
+    # check to see if pandas data frames are out of order,
+    # do sort here and re-check if data frames are different lengths?
+    if sample_df_metadata is not None:
+        if (sample_df.index != sample_df_metadata.index).any():
+            sample_df = sample_df.sort_index(axis=0)
+            sample_df_metadata = sample_df_metadata.sort_index(axis=0)
+
+    fig = plt.figure(figsize=(15, 15))
```

Comment:
This needs to be changed.  Maybe we can also pass in `figsize` as a `**kwargs` argument.

Also don't forget to check for `matplotlib.axes`